operations/terraform-state.yaml

```diff
@@ -0,0 +1,111 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+AWSTemplateFormatVersion: '2010-09-09'
+Description: 'Terraform state (S3 and DynamoDB)'
+Metadata:
+  'AWS::CloudFormation::Interface':
+    ParameterGroups:
+    - Label:
+        default: 'Parent Stacks'
+      Parameters:
+      - ParentKmsKeyStack
+    - Label:
+        default: 'Terraform State Parameters'
+      Parameters:
+      - TerrformStateIdentifier
+      - TerraformStateUserARNs
+      - TerraformStateAdminARNs
+Parameters:
+  ParentKmsKeyStack:
+    Description: 'Stack name of parent KMS key stack based on security/kms-key.yaml template.'
+    Type: String
+  TerrformStateIdentifier:
+    Description: 'Globally unique identifier for your Terraform state.'
+    Type: String
+  TerraformStateUserARNs:
+    Description: 'A comma delimited list of IAM users or role ARNs that need to use the Terraform state (see https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_notprincipal.html#specifying-notprincipal).'
+    Type: String
+  TerraformStateAdminARNs:
+    Description: 'A comma delimited list of IAM users or role ARNs that should be able to administer the Terraform state (see https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_notprincipal.html#specifying-notprincipal).'
+    Type: String
+Resources:
+  StateBucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+        - ServerSideEncryptionByDefault:
+            KMSMasterKeyID: {'Fn::ImportValue': !Sub '${ParentKmsKeyStack}-KeyArn'}
+            SSEAlgorithm: 'aws:kms'
+      BucketName: !Ref TerrformStateIdentifier
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+      VersioningConfiguration:
+        Status: 'Enabled'
+  StateBucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref StateBucket
+      PolicyDocument:
+        Statement:
+        - Effect: 'Deny'
+          Action: 's3:*'
+          Resource:
+          - !Sub '${StateBucket.Arn}'
+          - !Sub '${StateBucket.Arn}/*'
+          NotPrincipal:
+            AWS: !Split [',', {"Fn::Sub": '${TerraformStateUserARNs},${TerraformStateAdminARNs}'}]
+        - Effect: 'Deny'
+          NotAction:
+          - 's3:ListBucket'
+          - 's3:GetObject'
+          - 's3:PutObject'
+          Resource:
+          - !Sub '${StateBucket.Arn}'
+          - !Sub '${StateBucket.Arn}/*'
+          Principal:
+            AWS: !Split [',', !Ref TerraformStateUserARNs]
+  StateTable:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      AttributeDefinitions:
+      - AttributeName: 'LockID'
+        AttributeType: 'S'
+      BillingMode: 'PAY_PER_REQUEST'
+      KeySchema:
+      - AttributeName: 'LockID'
+        KeyType: HASH
+      SSESpecification:
```

Comment:
we could use the key from ParentKmsKeyStack?

Reply:
The DynamoDB table does not contain sensitive information.

Reply:
I see